base_optional_quick_create/model.py

```diff
@@ -50,14 +48,18 @@ def _register_hook(self, cr, ids=None):
                     model_obj.check_quick_create = True
         return True
 
+    def _register_hook(self, cr):
+        self._patch_quick_create(cr, self.search(cr, SUPERUSER_ID, []))
+        return super(IrModel, self)._register_hook(cr)
+
     def create(self, cr, uid, vals, context=None):
-        res_id = super(ir_model, self).create(cr, uid, vals, context=context)
-        self._register_hook(cr, [res_id])
+        res_id = super(IrModel, self).create(cr, uid, vals, context=context)
+        self._patch_quick_create(cr, [res_id])
         return res_id
 
     def write(self, cr, uid, ids, vals, context=None):
         if isinstance(ids, (int, long)):
             ids = [ids]
-        res = super(ir_model, self).write(cr, uid, ids, vals, context=context)
-        self._register_hook(cr, ids)
+        res = super(IrModel, self).write(cr, uid, ids, vals, context=context)
+        self._patch_quick_create(cr, ids)
```

Comment:
name_create should be patched when the model has avoid_quick_create set. Strictly speaking, patching should only take place when this field is being written.